frontend/javascripts/oxalis/merger_mode.js

```diff
@@ -0,0 +1,294 @@
+  }
+  if (didMappingChange) {
+    api.data.setMapping(mergerModeState.segmentationLayerName, mergerModeState.colorMapping);
+  }
+  call(action);
+}
+
+// Changes the opacity of the segmentation layer
+function changeOpacity(mergerModeState: MergerModeState) {
+  if (mergerModeState.segmentationOn) {
+    api.data.setConfiguration("segmentationOpacity", 0);
+    mergerModeState.segmentationOn = false;
+  } else {
+    api.data.setConfiguration("segmentationOpacity", mergerModeState.segmentationOpacity);
+    mergerModeState.segmentationOn = true;
+  }
+}
+
+function shuffleColorOfCurrentTree(mergerModeState: MergerModeState) {
+  const { treeColors, colorMapping, segmentationLayerName } = mergerModeState;
+  const setNewColorOfCurrentActiveTree = () => {
+    const activeTreeId = api.tracing.getActiveTreeId();
+    if (activeTreeId == null) {
+      Modal.info({ title: "Could not find an active tree." });
+      return;
+    }
+    const oldColor = getTreeColor(activeTreeId, mergerModeState);
+    // Reset the color of the active tree
+    treeColors[activeTreeId] = undefined;
+    // Applies the change of the color to all connected segments
+    Object.keys(colorMapping).forEach(key => {
+      if (colorMapping[key] === oldColor) {
+        colorMapping[key] = getTreeColor(activeTreeId, mergerModeState);
+      }
+    });
+    // Update the segmentation
+    api.data.setMapping(segmentationLayerName, colorMapping);
+  };
+
+  Modal.confirm({
+    title: "Do you want to set a new Color?",
+    onOk: setNewColorOfCurrentActiveTree,
+    onCancel() {},
+  });
+}
+
+async function mergeSegmentsOfAlreadyExistingTrees(index = 0, mergerModeState: MergerModeState) {
+  const { nodes, segmentationLayerName, nodeSegmentMap, colorMapping } = mergerModeState;
+  const numbOfNodes = nodes.length;
+  if (index >= numbOfNodes) {
+    return;
+  }
+
+  const [segMinVec, segMaxVec] = api.data.getBoundingBox(segmentationLayerName);
+
+  const setSegementationOfNode = async node => {
+    const pos = node.position;
+    const { treeId } = node;
+    // Skip nodes outside segmentation
+    if (
+      pos[0] < segMinVec[0] ||
+      pos[1] < segMinVec[1] ||
+      pos[2] < segMinVec[2] ||
+      pos[0] >= segMaxVec[0] ||
+      pos[1] >= segMaxVec[1] ||
+      pos[2] >= segMaxVec[2]
+    ) {
+      // The node is not in bounds of the segmentation
+      return;
+    }
+    const segmentId = await api.data.getDataValue(segmentationLayerName, pos);
+    if (segmentId != null && segmentId > 0) {
+      // Store the segment id
+      nodeSegmentMap[node.id] = segmentId;
+      // Add to agglomerate
+      increaseNodesOfSegment(segmentId, mergerModeState);
+      mapSegmentColorToTree(segmentId, treeId, mergerModeState);
+    }
+  };
+  const nodesMappedPromises = nodes.map(node => setSegementationOfNode(node));
+  await Promise.all(nodesMappedPromises);
+  api.data.setMapping(segmentationLayerName, colorMapping);
```

Comment:
Here was the original mistake: I used `"segmentation"` instead of `segmentationLayerName` which cause the merger mode not to work in hybrid tracing when having at least one node for the  preprocessing